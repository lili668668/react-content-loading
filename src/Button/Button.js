import React, { Component } from 'react'
import PropTypes from 'prop-types'

class Button extends Component {
  render () {
    const { children, onClick } = this.props
    return (
      <button onClick={onClick}>
        {children}
      </button>
    )
  }
}

Button.defaultProps = {
  children: '',
  onClick: () => {}
}

Button.propTypes = {
  /** The content in Button */
  children: PropTypes.any,
  /** The behavior when clicking Button */
  onClick: PropTypes.func
}

export default Button
