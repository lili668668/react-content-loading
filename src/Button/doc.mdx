---
name: Button
menu: Components
route: /components/button
---

import { Playground, PropsTable } from 'docz'
import Button from './Button'

# Button

## Description

A example component.

## Props
<PropsTable of={Button} />

## Basic usage

`import Button from 'react-component-lib-boilerplate/lib/Button'` or `import { Button } from 'react-component-lib-boilerplate'`

<Playground>
  <Button>Click Me</Button>
</Playground>
