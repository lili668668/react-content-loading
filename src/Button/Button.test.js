import React from 'react'
import { shallow } from 'enzyme'

import Button from './Button'

describe('Button', () => {
  it ('should render correctly', () => {
    const component = shallow(<Button />)
    expect(component).toMatchSnapshot()
  })

  it ('should not change anything when trigger default function', () => {
    const component = shallow(<Button />)
    component.simulate('click')
    expect(component).toMatchSnapshot()
  })
})

