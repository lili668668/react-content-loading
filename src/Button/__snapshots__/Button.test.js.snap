// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should not change anything when trigger default function 1`] = `
<button
  onClick={[Function]}
/>
`;

exports[`Button should render correctly 1`] = `
<button
  onClick={[Function]}
/>
`;
